Release Notes for 3.5.1

awholegunch opened the pull request:
Running a **Quick Build:** http://builds.cask.co/browse/CDAP-DQB126-4

Page of interest: **Release Notes:** http://builds.cask.co/artifact/CDAP-DQB126/shared/build-4/Docs-HTML/3.5.1-SNAPSHOT/en/reference-manual/release-notes.html#release-3-5-1

List of Release Note Issues: https://issues.cask.co/issues/?filter=11255

Commit:
First draft.

bdmogal commented:
https://issues.cask.co/browse/CDAP-7237 is missing.

Commit:
Updates.

Commit:
Fix link.

sagarkapare commented:
Please add https://issues.cask.co/browse/CDAP-7204

sagarkapare commented:
Need to add https://issues.cask.co/browse/HYDRATOR-912 as well.

Commit:
Adding issue 7179.

Commit:
Added issue HYDRATOR-912.

Commit:
Added issue 7204.

Commit:
Remove duplicate entry for issue 7179.

Commit:
Add details.

dereklwood commented on cdap-docs/reference-manual/source/release-notes.rst:
```diff
@@ -23,6 +23,106 @@ Cask Data Application Platform Release Notes
+
+- :cask-issue:`CDAP-7179` - In CDAP 3.5.0, new ``kafka.server.*`` properties replace older
+  properties such as ``kafka.log.dir``. Though the older properties were marked as
+  deprecated, they are in fact being ignored, and need to be replaced with the current
```
according to the bug, this is actually not true.  old properties in cdap-site.xml are not ignored.  The issue only comes into play when using CM, because the CSD might generate a cdap-site.xml with both old and new property names, in which the new property names win.  and this only pertains to the kafka.log.dir (vs kafka.server.log.dirs) property, since the CSD autogenerates a default for it using the new property name.

awholegunch replied:
Edited per your comments.

Commit:
Edit per comments.

dereklwood commented on cdap-docs/reference-manual/source/release-notes.rst:
```diff
@@ -41,16 +41,15 @@ Known Issues
+  <cloudera-installation-add-service-wizard-configuration>`, you need **to change** to the
+  new property ``kafka.server.log.dirs``, as the deprecated ``kafka.log.dir`` is being
+  ignored in favor of the new property. If you don't, your custom value will be replaced
+  with the default value of ``/tmp/kafka-logs``.
```
For CM, the default is actually `/var/tmp/cdap/kafka-logs` (auto-generated from CM's $LOCAL_DIR + "kafka-logs").  but maybe we can just say "the default" ?

awholegunch replied:
We can change this to "the default"; fine with me. But maybe we should add that somewhere else though? Should it be in `cdap-default.xml` (and then it would appear in the table) or in Cloudera Manager install notes, or both?

Commit:
Change description of "default value".

Commit:
Add placeholders for CDAP-7248 and CDAP-7249.

anwar6953 commented on cdap-docs/reference-manual/source/release-notes.rst:
```diff
@@ -23,6 +23,105 @@ Cask Data Application Platform Release Notes
+  completed. See the :ref:`Administration Manual: Authorization <admin-authorization>` for
+  details.)
+
+- :cask-issue:`CDAP-7179` - In CDAP 3.5.0, new ``kafka.server.*`` properties replace older
```
this is a known issue?

awholegunch replied:
Yes. It was also a 3.5.0 known issue.

anwar6953 commented on cdap-docs/reference-manual/source/release-notes.rst:
```diff
@@ -23,6 +23,105 @@ Cask Data Application Platform Release Notes
+
+
 `Release 3.5.0 <http://docs.cask.co/cdap/3.5.0/index.html>`__
 =============================================================
```
The known issue below should be modified

anwar6953 replied:
Maybe it should be changed to the known issue (regarding CM) in 3.5.1.

awholegunch replied:
Updated.

anwar6953 commented on cdap-docs/reference-manual/source/release-notes.rst:
```diff
@@ -23,6 +23,105 @@ Cask Data Application Platform Release Notes
+  
+  **If you are upgrading from CDAP 3.4.x to 3.5.x,** and you have set a value for
+  ``kafka.log.dir`` by using Cloudera Manager's :ref:`safety-valve mechanism
+  <cloudera-installation-add-service-wizard-configuration>`, you need **to change** to the
```
Why is "to change" in bold?

awholegunch replied:
No particular reason; just added emphasis. I can remove it.

awholegunch replied:
Fixed.

anwar6953 commented on cdap-docs/reference-manual/source/release-notes.rst:
```diff
@@ -23,6 +23,105 @@ Cask Data Application Platform Release Notes
+Bug Fixes
+---------
+
+- :cask-issue:`CDAP-3544` - Corrected an error in the documentation to correctly show 
```
It's not an error. The previous documentation simply showed example usage that is deprecated now.

awholegunch replied:
Removed.

Commit:
Removed emphasis from phrase.
Removed issue CDAP-3544 as it wasn't a big, and we don't normally show
documentation improvements.
Copied the content of the Known Issue 7179 of 3.5.1 to the same issue in
3.5.0.

Commit:
Added release notes for CDAP-7248 and CDAP-7249.

bdmogal commented on cdap-docs/reference-manual/source/release-notes.rst:
```diff
@@ -23,17 +23,121 @@ Cask Data Application Platform Release Notes
 Known Issues
 ------------
+
+- :cask-issue:`CDAP-7175` - Though an underlying issue has been fixed in the upgrade tool
```
this seems very verbose. Perhaps just refer to the doc on upgrade, instead of including it here again?

Also, does starting with "Though" seem to suggest the user is expected to have some prior knowledge about this?

awholegunch replied:
Edited and reduced in length.

Commit:
Removed CDAP-7224 as it has not been reproduced.

Commit:
Edited per comments. Re-flowed and tightened the second paragraph in this
commit.

Commit:
Merge branch 'release/3.5' into docs/r3.5.1_release_notes

Commit:
Further editing.

sagarkapare commented:
need to add https://issues.cask.co/browse/HYDRATOR-578

sagarkapare commented:
one comment. rest LGTM 👍

Commit:
Added issue HYDRATOR-578.

sagarkapare commented:
LGTM
